src/coreclr/vm/gchelpers.cpp

```diff
@@ -513,9 +513,11 @@ OBJECTREF TryAllocateFrozenSzArray(MethodTable* pArrayMT, INT32 cElements)
 
     // The initial validation is copied from AllocateSzArray impl
 
-    if (pArrayMT->ContainsPointers() && cElements > 0)
+    if (pArrayMT->Collectible() || (pArrayMT->ContainsPointers() && cElements > 0))
```

Comment:
```suggestion
    if (pArrayMT->ContainsPointers() && cElements > 0)
```
This change is unnecessary with the change in the JIT

Reply:
What check blocks collectible value types without this? Isn't the JIT check only for shared generics so reference types?

Reply:
Existing `CORJIT_FLAG_FROZEN_ALLOC_ALLOWED` check handles that.

Reply:
> Existing `CORJIT_FLAG_FROZEN_ALLOC_ALLOWED` check handles that.

I thought that'd only check if the type we're currently compiling is in a collectible assembly, not if its generic parameters are.

Reply:
`CORJIT_FLAG_FROZEN_ALLOC_ALLOWED` covers generic parameters as well.